server/src/main/java/org/elasticsearch/common/util/BytesRefHash.java

```diff
@@ -29,7 +29,7 @@ public final class BytesRefHash extends AbstractHash implements Accountable {
         // spare BytesRef
         + RamUsageEstimator.shallowSizeOfInstance(BytesRef.class);
 
-    private final BytesRefArray bytesRefs;
+    private BytesRefArray bytesRefs;
```

Comment:
I can't quite see why we removed the final from this field definition here? Also related question, can we make hashes final too?